### ISSUE TITLE
no need to explicitly initialize the model layers for training from scratch or even for fine-tuning?

### ISSUE THREAD

**Issue by love2learn-7179** [Tianxiaomo/pytorch-YOLOv4#67]
We don't need to explicitly initialize the model layers? sth like this @https://pytorch.org/docs/stable/_modules/torchvision/models/resnet.html#resnet18

        for m in self.modules():
            if isinstance(m, nn.Conv2d):
                nn.init.kaiming_normal_(m.weight, mode='fan_out', nonlinearity='relu')
            elif isinstance(m, (nn.BatchNorm2d, nn.GroupNorm)):
                nn.init.constant_(m.weight, 1)
                nn.init.constant_(m.bias, 0)

BTW, the default initialization for conv2d is following @https://discuss.pytorch.org/t/what-is-the-default-initialization-of-a-conv2d-layer-and-linear-layer/16055

    def reset_parameters(self):
        n = self.in_channels
        for k in self.kernel_size:
            n *= k
        stdv = 1. / math.sqrt(n)
        self.weight.data.uniform_(-stdv, stdv)
        if self.bias is not None:
            self.bias.data.uniform_(-stdv, stdv)

**Comment by Tianxiaomo** [comment]
https://github.com/AlexeyAB/darknet/blob/e6469eb071521a4ff5be8c0e9cceb524d0a65a95/src/convolutional_layer.c#L445

https://github.com/AlexeyAB/darknet/blob/e6469eb071521a4ff5be8c0e9cceb524d0a65a95/src/convolutional_layer.c#L705

This is the conv initialization implementation of darknet.
